captainPrime/Politico
# #163388914 Build REST API to create a party

**What does this PR do?**
Implement RESTFUL API to create a political party

**Description of Task to be completed?**
the POST request method (POST/api/v1/parties) requests that a web server accepts the data enclosed in the body of the request message which is the offices in a JSON file, the request is received and stored in as an array contain the details of the parties
have the endpoint working:

- `POST /api/v1/offices`

- `GET /api/v1/offices`

- `GET /api/v1/parties`

- `GET /api/v1/parties`

- `GET /api/v1/offices/<office-id>`

- `GET  /api/v1/parties/<party-id>`

- `DELETE  /api/v1/<party-id>`

- `PATCH /api/v1/<party-id>/name`

**How should this be manually tested?**

`step 1`
- To run this app, create  a folder and clone the repo,
`git clone https://github.com/captainPrime/Politico.git` 
- after repo has been cloned, pull branch 
`git pull origin <branch name>` into folder. 
this creates a replica of this gitHub repo. Ensure the the following are installed on the system: node.js, express.js, eslint, mocha ...

`step 2`
- cd into the repo

`step 3`
- To run endpoints type in command `npm start`
- This shows a log `loading port 3000.....`

`step 4`
- After, run specified api - links and send on postman to view results

`POST http://localhost:3000/api/v1/parties/`

**For Test driven development**
Run command `npm run test` 

**Any background context you want to provide**

none

**What are the relevant pivotal tracker stories**
https://www.pivotaltracker.com/story/show/163388914
#163388914
